main.go

```diff
@@ -183,7 +183,12 @@ func visit(url *url.URL) {
 		log.Fatalf("unable to create request: %v", err)
 	}
 	for _, h := range httpHeaders {
-		req.Header.Add(headerKeyValue(h))
+		k, v := headerKeyValue(h)
+		if strings.EqualFold(k, "host") {
```

Comment:
I can't think of a case where you would need to use `EqualFold()` for an HTTP header field-name. They should only be ASCII-compatible bytes.